src/lib.rs

```diff
@@ -209,7 +295,7 @@ impl Connection<NotConnected> {
     /// # }
     /// ```
     pub fn with_token(self, token: &str) -> Connection<Authenticated> {
-        let mut opts = self.options.clone();
```

Comment:
No need to clone the backing options because this is a `self` function that takes ownership over them anyway.

Reply:
+1